src/WindowManager.vala

```diff
@@ -723,19 +723,19 @@ namespace Gala {
         }
 
         private void dim_parent_window (Meta.Window window, bool dim) {
-            if (window.window_type == Meta.WindowType.MODAL_DIALOG) {
-                var ancestor = window.find_root_ancestor ();
-                if (ancestor != null && ancestor != window) {
-                    var win = (Meta.WindowActor) ancestor.get_compositor_private ();
-                    // Can't rely on win.has_effects since other effects could be applied
-                    if (dim) {
+            unowned var ancestor = window.find_root_ancestor ();
+            if (ancestor != null && ancestor != window) {
+                unowned var win = (Meta.WindowActor) ancestor.get_compositor_private ();
+                // Can't rely on win.has_effects since other effects could be applied
+                if (dim) {
+                    if (window.window_type == Meta.WindowType.MODAL_DIALOG) {
                         var dark_effect = new Clutter.BrightnessContrastEffect ();
                         dark_effect.set_brightness (-0.4f);
 
                         win.add_effect (dark_effect);
-                    } else {
-                        win.clear_effects ();
                     }
+                } else {
+                    win.clear_effects ();
```

Comment:
Since this function only applies the `dark_effect` to `MODAL_DIALOG` windows if `dim` is `true`,  should it not disapply only that effect to only the same window type when `dim` is `false`?   As you say, other effects could be applied to other windows.

Reply:
@jeremypw possibly? I just know that with master I can reliably reproduce an issue where the "Unsaved changes" dialog in Code will add a darken effect, but when the window closes the effect is not removed. And this branch fixes it for me